### ISSUE TITLE
Creating Money struct from german formatted string

### ISSUE THREAD

**Issue by f34nk** [kipcole9/money#83]
I get a conversion error for a german formatted number string.

`Money.new("1.000,99", :EUR)`

`{:error,
 {Money.InvalidAmountError,
  "Amount cannot be converted to a number: \"1.000,99\""}}`

Any ideas how to solve this?

Frank

**Comment by kipcole9** [comment]
@f34nk Thanks for the issue.  `Money.new/2` when given a `String` amount will delegate to `Decimal.new/1`.  And that function accepts numbers formatted the same way as a numeric literal in Elixir.  That is, the grouping character is `_` and the decimal separator is `.`.

[CLDR](http://unicode.org/reports/tr35/tr35-numbers.html) does provide a specification tion for parsing numbers in a locale-specific way but that is not implemented in [ex_cldr](https://hex.pm/packages/ex_cldr) and its not a high priority for me.

Therefore I regret that the immediate solution to your situation would require transforming the string number literal into one that is acceptable to `Decimal.new/1`.

## Possible approach

I could implement a mechanism to *automate* that in a locale-specific fashion.  Such an approach would:

1. Remove all instances of `:group` and 
2. Replace any instance of `:decimal` with a fixed `.`

For the "de" locale the representative data would be:

```elixir
%{
  latn: %{
    decimal: ",", 
    exponential: "E",
    group: ".",
    infinity: "∞",
    list: ";", 
    minus_sign: "-",
    nan: "NaN",
    per_mille: "‰",
    percent_sign: "%",
    plus_sign: "+",
    superscripting_exponent: "·",
    time_separator: ":"
  }
}
```

### Questions:
1. Would that approach be useful for you (and any others following this issue)
2. Should this happen by default or should this be an option to `Money.new/2`

**Comment by f34nk** [comment]
Thank you for the quick reply.

To answer your questions:
My use case is parsing account transactions from CSV of a specific german bank. So, yes, I believe creating Money structs from locale-specific formatting would be very useful.

To maintain flexibility I would prefer it as an option to `Money.new/2`.

`Money.new("1.000,99", :EUR, locale: "de")`
`#Money<:EUR, 1000.99>`

**Comment by kipcole9** [comment]
I have added an `options` param to the `Money.new/2` function which is now `Money.new/3`.  The only valid option is `:locale`.  So per your original issue, the following now works:
```elixir
iex> Money.new("1.000,99", :EUR, locale: "de")
#Money<:EUR, 1000.99>
```
Would you consider testing this new version before I release it?  You would need to change your deps to include it from GitHub: `{:ex_money, git: "https://github.com/kipcole9/money"}`

All tests pass so I think its good to go but your validation would be helpful.

**Comment by kipcole9** [comment]
Published on hex as [version 2.10.0](https://hex.pm/packages/ex_money).
